[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# prefer-unified-property-style

Discourage against mixing atomic and composite forms of the same property in a style declaration. Atomic styles give more consistent results

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const color = 'red.100';
const styles = css({ borderRadius:"lg", borderTopRightRadius: "0" });
```
```js

import { Circle } from './panda/jsx';

function App(){
  const bool = true;
  return <Circle border="solid 1px" borderColor="gray.800" />;
}
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ borderColor: 'gray.900', borderWidth: '1px' });
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle marginTop="2" marginRight="3" />;
}
```

## Resources

* [Rule source](/plugin/src/rules/prefer-unified-property-style.ts)
* [Test source](/tests/prefer-unified-property-style.test.ts)
